pkg/pkg.bzl

```diff
@@ -379,3 +379,50 @@ def pkg_deb(name, package, **kwargs):
         changes = out_changes,
         **kwargs
     )
+
+def _pkg_zip_impl(ctx):
+  # Compute the relative path
+  data_path = compute_data_path(ctx.outputs.out, ctx.attr.strip_prefix)
+
+  args = ctx.actions.args()
+
+  for f in ctx.files.srcs:
+    args.add("--file", "%s=%s" % (_quote(f.path), dest_path(f, data_path)))
+
+  args.add("--output", ctx.outputs.out)
+  args.use_param_file("%s")
+
+  ctx.actions.run(
+      mnemonic = "PackageZip",
+      inputs = ctx.files.srcs,
+      executable = ctx.executable.build_zip,
+      arguments = [args],
+      outputs = [ctx.outputs.out],
+      env = {
+          "LANG": "en_US.UTF-8",
+          "LC_CTYPE": "UTF-8",
+          "PYTHONIOENCODING": "UTF-8",
+          "PYTHONUTF8": "1",
+      },
+      use_default_shell_env = True,
+  )
+  return OutputGroupInfo(out=[ctx.outputs.out]);
+
+pkg_zip = rule(
+    implementation = _pkg_zip_impl,
+    attrs = {
+        "extension": attr.string(default = "zip"),
+        "strip_prefix": attr.string(),
+        "srcs": attr.label_list(allow_files = True),
+        # Implicit dependencies.
+        "build_zip": attr.label(
+            default = Label("@rules_pkg//:build_zip"),
+            cfg = "host",
+            executable = True,
+            allow_files = True,
+        ),
+    },
+    outputs = {
```

Comment:
Outputs is going away. This needs to have an explicit attr.output() and return an OutputGroupInfo. See https://github.com/bazelbuild/rules_pkg/commit/30dd90482fd2387d9641eaca37b5c303b2e3fb36
for an example of the change

Reply:
Opps I forgot incorporate all the changes into that last PR, one sec.

Reply:
There we go